src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java

```diff
@@ -148,6 +151,26 @@ public ResponseEntity deleteTeamPermissions(@PathVariable String teamUUID, @Requ
         return ResponseEntity.ok(teams.stream().map(TeamDto::from).collect(Collectors.toSet()));
     }
 
+    @GetMapping(value = "/team/users")
+    public ResponseEntity<Set<UserWithTeamsDto>> getUsersForTeam() {
+        List<UserDto> users = userService.getAllUsers();
+        Set<UserWithTeamsDto> usersWithTeams = new HashSet<>();
+
+        for (UserDto user: users){
+            Set<Team> teams = teamService.getTeamsForUser(UUID.fromString(user.getId()));
+            ArrayList<String> teamNames = new ArrayList<>();
```

Comment:
While this works due to them both being incremented at the same time - wouldn't a Map<String of Unit, List<String of Teams>> be more applicable?

Reply:
Resolved to add further comments.